bitcoin/src/merkle_tree.rs

```diff
@@ -19,13 +26,13 @@ use crate::consensus::encode::Encodable;
 
 /// Calculates the merkle root of a list of *hashes*, inline (in place) in `hashes`.
 ///
-/// In most cases, you'll want to use [bitcoin_merkle_root] instead.
+/// In most cases, you'll want to use [`calculate_root`] instead.
 ///
 /// # Returns
 /// - `None` if `hashes` is empty. The merkle root of an empty tree of hashes is undefined.
 /// - `Some(hash)` if `hashes` contains one element. A single hash is by definition the merkle root.
 /// - `Some(merkle_root)` if length of `hashes` is greater than one.
-pub fn bitcoin_merkle_root_inline<T>(hashes: &mut [T]) -> Option<T>
+pub fn calculate_root_inline<T>(hashes: &mut [T]) -> Option<T>
```

Comment:
Now that I see this I realize it'd be nice to explicitly state that the hashes are overwritten in the process.  And perhaps state that the state we leave it in is undefined and must not be relied on.

Reply:
Agreed. Though FWIW taking a mutable slice implies that we'll overwrite stuff.

Reply:
Added some extra docs as an additional patch.